apps/frontend/src/app/PageTeam/CharacterDisplay/Tabs/TabOptimize/Components/ExcludeLoadout.tsx

```diff
@@ -0,0 +1,55 @@
+import { useDatabase, useOptConfig } from '@genshin-optimizer/gi/db-ui'
+import { CheckBox, CheckBoxOutlineBlank } from '@mui/icons-material'
+import { Button } from '@mui/material'
+import { useContext, useEffect } from 'react'
+import { TeamCharacterContext } from '../../../../../Context/TeamCharacterContext'
+
+export default function ExcludeLoadout({
+  disabled = false,
+}: {
+  disabled?: boolean
+}) {
+  const database = useDatabase()
+  const {
+    teamCharId,
+    teamChar: { optConfigId },
+    team,
+  } = useContext(TeamCharacterContext)
+  const { useLoadoutExclusion } = useOptConfig(optConfigId)
+
+  useEffect(() => {
+    console.log('Update ExcludeLoadout')
+    const chars = team.teamCharIds
+      .filter((id) => id !== teamCharId)
+      .map((id) => database.teamChars.get(id))
+    const buildIds = chars.map((char) => char.buildIds).flat()
```

Comment:
This excludes "every artifact in every build for every character", the feature is intended to exclude "every artifact in the **active** build for every character".

You can approach this via

```
const artIdsToExclude:string[] = team.teamCharIds
      .filter((id) => id !== teamCharId)
      .flatmap((id) => database.teamChars.getLoadoutArtifacts(id))
      .map((art:?ICachedArtifact)=>art?.id)
      .filter(id=>!!id)
 ```

Reply:
admittedly, I should name this function to `getActiveBuildArtifacts` to be more appropriate.